### ISSUE TITLE
[Bug]: Screen reader is announcing incorrect role as spin button for sale control in Windows.

### ISSUE THREAD

**Issue by msftedad** [ceolter/ag-grid#9944]
### Link to reproducible scenario

main.ts - nodebox - CodeSandbox

### Describe the bug

Steps to reproduce 

1. Launch #URL: [https://codesandbox.io/p/sandbox/number-filter-forked-ks5gxn?file=%2fmain.ts%3A53%2C28](https://codesandbox.io/p/sandbox/number-filter-forked-ks5gxn?file=%2fmain.ts:53%2c28)
2. Press Tab key till to reach second row sale control edit fields.
3. Observe on using tab key, whether Screen reader is announcing incorrect role as spin button for sale control or not.

Actual behaviour
Screen reader is announcing incorrect role as spin button for sale control in Windows. Screen reader is announcing as Sale ($) Filter Input spin button editable blank.

https://github.com/user-attachments/assets/5a24653c-bf2e-413b-9283-6b19f0cb0767

<img width="960" alt="Image" src="https://github.com/user-attachments/assets/edd1e8a1-50b9-4860-aaea-9b3128cc5d7d" />

Expected behaviour
Screen reader should not announce incorrect role as spin button for sale control in Windows.
The edit fields should have the correct role assigned, which accurately represents their functionality as standard input fields, not spin buttons.

More information
This issue is observed with all three-screen readers.
The Number filter edit fields present in the second row are assigned the role of a spin button, even though they are standard edit fields.

### Version

31.3.4

### Does the issue occur for a specific framework only?

All languages

### Is the issue only observable on a specific browser?

All browsers

**Comment by AG-Zoheil** [comment]
Hello 

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference: **AG-14697**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

You can follow the status of this item on the product pipeline page:
https://ag-grid.com/pipeline/

The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Thanks again for bringing this up with us.

Kind regards,
Zoheil
